03-rnaseq/gene-id-annotation_rnaseq_01_ensembl.Rmd

```diff
@@ -0,0 +1,380 @@
+However, the data is now in a `list` object, making it a little more difficult to explore.
+We are going to turn our list object into a data frame object in the next chunk.
+
+```{r}
+# Let's make our object a bit more manageable for exploration by turning it into a data frame
+mapped_df <- mapped_list %>%
+  tibble::enframe(name = "Ensembl", value = "Entrez") %>%
+  tidyr::unnest() # enframe makes a `list` column, so here we are converting that to simpler format
+```
+
+Now let's take a peek at our data frame.
+
+```{r}
+head(mapped_df)
+```
+
+We can see that our data frame has a new column `Entrez`.
+Let's get a summary of the values returned in the `Entrez` column of our mapped data frame.
+
+```{r}
+# We can use the `summary()` function to get a better idea of the distribution of values in the `Entrez` column
+summary(as.factor(mapped_df$Entrez)) # We need to use the `as.factor()` function here in order to get the count of each unique value returned
+```
+
+There are 9102 NA's in our data frame, which means that 9102 out of the 31882 Ensembl IDs did not map to Entrez IDs.
+This is not the best rate, in our opinion, which means that it may be wise to try a different gene identifier type that may have a better mapping rate.
+Regardless, it is always good to be aware of how many genes you are potentially "losing" if you rely on this new gene identifier you've mapped to for downstream analyses.
+
+Now let's check to see how many genes we have that were mapped to multiple IDs.
+
+```{r}
+multi_mapped_df <- mapped_df %>%
+  # Let's count the number of times each Ensembl ID appears in `Ensembl` column
+  dplyr::count(Ensembl, name = "entrez_id_count") %>%
+  # Arrange by the genes with the highest number of Entrez IDs mapped
+  dplyr::arrange(desc(entrez_id_count))
+
+# Let's look at the first 6 rows of our `multi_mapped` object
+head(multi_mapped_df)
+```
+
+Looks like we have some cases where 15 Entrez IDs mapped to a single Ensembl ID.
+We have a total of 235 out of 31882 Ensembl IDs with multiple mappings to Entrez IDs.
+With that rate, we can filter out the multi-mapped IDs if we want to keep 1:1 mappings for our downstream analyses by supplying the `"filter"` option to the `multiVals` argument of `mapIds()`.
+If you want to see an example of a `multiVals = "filter"` strategy see the [microarray gene identifier annotation example notebook](https://github.com/AlexsLemonade/refinebio-examples/tree/master/02-microarray/gene-id-annotation_microarray_01_ensembl.html).
+
+### Store all mapped values
+
+We are not sure which Entrez ID is relevant as yet, but we could store all of the mapped information for future use.
+In the next chunk, we show how we can collapse all the Entrez IDs into one column where we store them all for each unique Ensembl ID.
+
+```{r}
+final_mapped_df <- df %>%
+  # Collapse the mapped values in `mapped_list` into one column named `entrez_ids`
+  dplyr::mutate(entrez_ids = sapply(mapped_list, paste, collapse = ";")) %>%
```

Comment:
```suggestion
  dplyr::mutate(all_entrez_ids = sapply(mapped_list, paste, collapse = ";")) %>%
```
Since below you named the next column, `first_entrez_ids`, you can name this one in parallel. (If you do this you'll need to change the next couple lines too.

Reply:
Including the comments!